### PR TITLE
feat: Add FAQ link to unknown pod watch error. Fixes #6886

### LINKED ISSUE #6886: more helpful error than `unknown (watch pods)` when workflow role is set-up incorrectly
Time and time again, I find myself telling users this is actually a permission issue. It is usually accompanied by `Watch pods 403'. We could easily improve this experience by printing a link to [the FAQ](https://github.com/argoproj/argo-workflows/blob/master/docs/faq.md), and then making sure the FAQ has some helpful text.

### PR DESCRIPTION
**Pull request by momom-i**
Signed-off-by: momom <51011095+momom-i@users.noreply.github.com>

Tips:

* Maybe add you organization to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).
* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it does not need to pass
* Sign-off your commits to pass the DCO check: `git commit --signoff`.
* Run `make pre-commit -B` to fix codegen or lint problems. 
* Say how how you tested your changes. If you changed the UI, attach screenshots.
* If changes were requested, and you've made them, then dismis the review to get it looked at again.
* You can ask for help! 

Fix #6886

### CODE REVIEW

**Comment by momom-i** [comment]
When I set wrong permission service account, only `unknown (get pods)` error occurred instead of `unknown (watch pods)`. 
So I checked if `unknown (get pods)` or not.

File: workflow/executor/k8sapi/client.go [hunk 1]
```diff
@@ -69,6 +70,9 @@ func (c *k8sAPIClient) getPod(ctx context.Context) (*corev1.Pod, error) {
 	err := waitutil.Backoff(backoffOver30s, func() (bool, error) {
 		var err error
 		pod, err = c.clientset.CoreV1().Pods(c.namespace).Get(ctx, c.podName, metav1.GetOptions{})
+		if err != nil && strings.Contains(err.Error(), "unknown (get pods)") {
```
Comment (alexec) [inline workflow/executor/k8sapi/client.go:73]: can we create a func return `errWithHelp(err) err`
then we can 

```golang
return !errorsutil.IsTransientErr(err), errWithHelp(err)
````

Comment (momom-i) [reply workflow/executor/k8sapi/client.go:77]: thank you, fixed 410c2141b5edd9b14972d92abdc6bd68aac07798

**Review by alexec — APPROVED** [review]

**Review by alexec — APPROVED** [review]

**Review by alexec — CHANGES_REQUESTED** [review]
Can you sync with master so that the builds run please?

File: workflow/executor/k8sapi/client.go (hunk 1 above)
Comment (alexec) [reply workflow/executor/k8sapi/client.go:77]: can we put `strings.Contains(err.Error(), "unknown (get pods)")` into own func? e.g. `isErrUnknownGetPods`

**Review by alexec — CHANGES_REQUESTED** [review]

**Review by alexec — APPROVED** [review]

**Comment by momom-i** [comment]
I don't understand why CI fails...

**Comment by alexec** [comment]
Trying syncing with master?

File: workflow/executor/k8sapi/client.go (hunk 1 above)
Comment (alexec) [reply workflow/executor/k8sapi/client.go:81]: still need updating?

**Comment by momom-i** [comment]
I tried to sync with master again, but they seemed fail.

**Pull request merged**